components/WalletConnectButton/WalletConnectButton.tsx

```diff
@@ -4,8 +4,12 @@ import Image from "next/image"
 import { FC } from "react"
 import customLoader from "../../lib/customLoader"
 
+interface IWalletConnectFuncChild {
+  openConnectModal: () => void
+}
+
 interface WalletConnectButtonProps {
-  children?: React.ReactNode
+  children?: (props: IWalletConnectFuncChild) => any | React.ReactNode
```

Comment:
can you make this prop  `openConnectModal` `optional` so you don't need to update all existing implementations?